apps/antalmanac/src/components/RightPane/SectionTable/PrereqTree.tsx

```diff
@@ -40,30 +46,37 @@ interface TreeProps {
 
 const Tree: FC<TreeProps> = (props) => {
     // eslint-disable-next-line prefer-const
-    let prerequisite = props.prerequisite;
-    let isValueNode = typeof prerequisite === 'string';
+    const prerequisite = props.prerequisite;
+    const isValueNode = 'courseId' in prerequisite || 'examName' in prerequisite;
 
-    // if value is a string, render leaf node
     if (isValueNode) {
         return (
             <li key={props.index} className={'prerequisite-node'}>
-                <Node label={prerequisite as string} node={'prerequisite-node'} />
+                <Node
+                    label={`${prerequisite.courseId ?? prerequisite.examName ?? ''}${
+                        prerequisite?.minGrade ? ` (min grade = ${prerequisite?.minGrade})` : ''
+                    }${prerequisite?.coreq ? ' (coreq)' : ''}`}
+                    node={'prerequisite-node'}
+                />
             </li>
         );
-    }
-    // if value is an object, render the rest of the sub tree
-    else {
+    } else {
+        const prereqTree = prerequisite as Record<string, PrerequisiteNode[]>;
         return (
             <div className={'prerequisite-node'}>
                 <div style={{ display: 'inline-flex', flexDirection: 'row', padding: '0.5rem 0' }}>
                     <span style={{ margin: 'auto' }}>
                         <div className={'prereq-branch'}>
-                            {Object.prototype.hasOwnProperty.call(prerequisite, 'OR') ? 'one of' : 'all of'}
+                            {
+                                Object.entries(phraseMapping).filter(([x, _]) =>
```

Comment:
super-nit: one-letter variable name `x`. Maybe do something like `propertyName`. I'm even having trouble coming up with something because I barely understand what `x` is supposed to be here.